src/js/angular/core/directives/queryeditor/query-editor.directive.js

```diff
@@ -574,21 +583,46 @@ function queryEditorDirective($timeout, $location, toastr, $repositories, Sparql
             window.yasr = yasr;
             yasr.afterCopy = afterCopy;
             yasr.getQueryResultsAsFormat = function (downloadFormat) {
-                // Simple cross-browser download with a form
-                const $wbDownload = $('#wb-download');
-                $wbDownload.attr('action', 'repositories/' + $repositories.getActiveRepository());
-                $('#wb-download-query').val(scope.currentQuery.query);
-                if (window.editor.getValue() !== scope.currentQuery.query) {
+                // JSON and JSON-LD have to be fetched in memory before sending, so error handling must be introduced
+                // Valid for construct queries only
+                const exportTypesToFetchWithErrorHandling = ['application/rdf+json', 'application/ld+json'];
+                const isConstructQuery = window.editor.getQueryType() === 'CONSTRUCT';
+                const queryParams = {
+                    query: scope.currentQuery.query,
+                    infer: scope.currentQuery.inference,
+                    sameAs: scope.currentQuery.sameAs,
+                    auth: localStorage.getItem('com.ontotext.graphdb.auth')
+                };
+                const accept = downloadFormat;
+
+                const fetchWithErrorHandling = exportTypesToFetchWithErrorHandling.includes(downloadFormat);
+                if (window.editor.getValue() !== queryParams.query) {
                     toastr.warning($translate.instant('query.editor.query.results.mismatch'));
                 }
-                $('#wb-download-infer').val(scope.currentQuery.inference);
-                $('#wb-download-sameAs').val(scope.currentQuery.sameAs);
-                const auth = localStorage.getItem('com.ontotext.graphdb.auth');
-                if (auth) {
-                    $('#wb-auth-token').val(auth);
+                if (isConstructQuery && fetchWithErrorHandling) {
+                    RDF4JRepositoriesRestService.downloadResultsAsFile($repositories.getActiveRepository(), queryParams, accept)
+                        .then(function ({data, filename}) {
+                            saveAs(data, filename);
+                        }).catch(function (res) {
+                        // data is received as blob
+                        res.data.text()
+                            .then((message) => {
+                                toastr.error(message, $translate.instant('common.error'));
+                            });
+                    });
+                } else {
+                    // Simple cross-browser download with a form
```

Comment:
The setting of `wb-download-` val is a good candidate for an util method. You can create one if is used somewhere else.

Reply:
No sure if used anyware else,, won't do.